books/free-programming-books.md

```diff
@@ -3065,7 +3070,7 @@ Kerridge (PDF) (email address *requested*, not required)
 
 ### Windows Phone
 
-* [Developing An Advanced Windows Phone 7.5 App That Connects To The Cloud](http://coolthingoftheday.blogspot.co.uk/2012/05/free-ebook-guidance-advanced-windows.html)
+* [Developing An Advanced Windows Phone 7.5 App That Connects To The Cloud](https://web.archive.org/web/20150709045622/http://download.microsoft.com/download/C/4/6/C4635738-5E06-4DF7-904E-BDC22AED2E58/Developing%20an%20Advanced%20Windows%20Phone%207.5%20App%20that%20Connects%20to%20the%20Cloud.pdf) (PDF)
```

Comment:
PDF format is enforced by this rule: https://github.com/vhf/remark-lint-books-links

```suggestion
* [Developing An Advanced Windows Phone 7.5 App That Connects To The Cloud](https://msdn.microsoft.com/en-us/library/gg490765.aspx) - MSDN Library, David Britch, Francis Cheung, Adam Kinney, Rohit Sharma (HTML, [PDF](https://web.archive.org/web/20150709045622/http://download.microsoft.com/download/C/4/6/C4635738-5E06-4DF7-904E-BDC22AED2E58/Developing%20an%20Advanced%20Windows%20Phone%207.5%20App%20that%20Connects%20to%20the%20Cloud.pdf), code samples) (:card_file_box: *archived*)
```
This above patch should works now with extra info preservation 🤗

Try it 😉

Reply:
It passed all the checks.